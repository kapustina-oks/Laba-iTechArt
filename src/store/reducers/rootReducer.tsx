import { combineReducers } from "redux";
import authReducer from "./authReducer";
import adminReducer from "./adminReducer";
import cartReducer from "./cartReducer";

const rootReducer = combineReducers({
  auth: authReducer,
  admin: adminReducer,
  cart: cartReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export default rootReducer;
