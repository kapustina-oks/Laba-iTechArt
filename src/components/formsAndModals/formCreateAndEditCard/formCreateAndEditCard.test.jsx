import FormCreateAndEditCard from "./formCreateAndEditCard";
import { fireEvent, render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { describe } from "jest";
import rootReducer from "../../../store/reducers/rootReducer";

export const { expect } = window;

describe("FormCreateAndEditCard tests", () => {
  const store = configureStore({ reducer: rootReducer });

  it("render FormCreateAndEditCard without props", () => {
    render(
      <Provider store={store}>
        <FormCreateAndEditCard />
      </Provider>
    );
  });

  it("snapshot FormCreateAndEditCard", () => {
    render(
      <Provider store={store}>
        <FormCreateAndEditCard />
      </Provider>
    );
    expect(document.body).toMatchSnapshot();
  });

  it("render checkbox", () => {
    const handleHasCategoryChange = jest.fn();
    const { container } = render(
      <input type="checkbox" className="checkbox" id="playstation" onChange={handleHasCategoryChange("hasXbox")} />
    );
    const checkbox = container.firstChild;
    expect(checkbox).not.toBeChecked();
    fireEvent.click(checkbox);
    expect(handleHasCategoryChange).toHaveBeenCalledTimes(1);
    expect(checkbox).toBeChecked();
  });
});
