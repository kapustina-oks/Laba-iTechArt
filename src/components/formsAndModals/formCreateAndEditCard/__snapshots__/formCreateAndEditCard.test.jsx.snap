// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCreateAndEditCard tests render FormCreateAndEditCard without props 1`] = `
<body>
  <div>
    <form
      class="product-form"
    >
      <div
        class="flex-product"
      >
        <div>
          <img
            alt="game_img"
            class="size_img"
            src="https://media.contentapi.ea.com/content/dam/gin/images/2021/06/battlefield-2042-key-art.jpg.adapt.crop1x1.767p.jpg"
          />
        </div>
        <div
          class="input-product-group"
        >
          <label
            class="label-product"
          >
            Name
            <input
              name="name"
              type="text"
              value=""
            />
          </label>
          <label
            class="label-product"
          >
            Genre
            <input
              name="genres"
              type="text"
              value=""
            />
          </label>
          <label
            class="label-product"
          >
            Price
            <input
              name="price"
              type="text"
              value="0$"
            />
          </label>
          <label
            class="label-product"
          >
            Img
            <input
              name="img"
              type="text"
              value="https://media.contentapi.ea.com/content/dam/gin/images/2021/06/battlefield-2042-key-art.jpg.adapt.crop1x1.767p.jpg"
            />
          </label>
        </div>
      </div>
      <label
        class="label-product"
      >
        Description
        <textarea
          class="description-textarea"
          name="description"
        />
      </label>
      <label
        class="label-product"
      >
        Age
        <select
          class="select-product"
          name="select"
        >
          <option
            selected=""
            value="6+"
          >
            6+
          </option>
          <option
            value="12+"
          >
            12+
          </option>
          <option
            value="18+"
          >
            18+
          </option>
        </select>
      </label>
      <label
        class="label-product"
      >
        Xbox
        <input
          class="checkbox"
          id="xbox"
          type="checkbox"
        />
      </label>
      <label
        class="label-product"
      >
        PC
        <input
          class="checkbox"
          id="pc"
          type="checkbox"
        />
      </label>
      <label
        class="label-product"
      >
        Playstation
        <input
          class="checkbox"
          id="playstation"
          type="checkbox"
        />
      </label>
      <div
        class="btn-product-group"
      >
        <button
          class="btn-product"
          type="submit"
        >
          Submit
        </button>
        <button
          class="btn-product"
          type="submit"
        >
          Delete Card
        </button>
      </div>
    </form>
  </div>
</body>
`;

exports[`FormCreateAndEditCard tests snapshot FormCreateAndEditCard 1`] = `
<body>
  <div>
    <form
      class="product-form"
    >
      <div
        class="flex-product"
      >
        <div>
          <img
            alt="game_img"
            class="size_img"
            src="https://media.contentapi.ea.com/content/dam/gin/images/2021/06/battlefield-2042-key-art.jpg.adapt.crop1x1.767p.jpg"
          />
        </div>
        <div
          class="input-product-group"
        >
          <label
            class="label-product"
          >
            Name
            <input
              name="name"
              type="text"
              value=""
            />
          </label>
          <label
            class="label-product"
          >
            Genre
            <input
              name="genres"
              type="text"
              value=""
            />
          </label>
          <label
            class="label-product"
          >
            Price
            <input
              name="price"
              type="text"
              value="0$"
            />
          </label>
          <label
            class="label-product"
          >
            Img
            <input
              name="img"
              type="text"
              value="https://media.contentapi.ea.com/content/dam/gin/images/2021/06/battlefield-2042-key-art.jpg.adapt.crop1x1.767p.jpg"
            />
          </label>
        </div>
      </div>
      <label
        class="label-product"
      >
        Description
        <textarea
          class="description-textarea"
          name="description"
        />
      </label>
      <label
        class="label-product"
      >
        Age
        <select
          class="select-product"
          name="select"
        >
          <option
            selected=""
            value="6+"
          >
            6+
          </option>
          <option
            value="12+"
          >
            12+
          </option>
          <option
            value="18+"
          >
            18+
          </option>
        </select>
      </label>
      <label
        class="label-product"
      >
        Xbox
        <input
          class="checkbox"
          id="xbox"
          type="checkbox"
        />
      </label>
      <label
        class="label-product"
      >
        PC
        <input
          class="checkbox"
          id="pc"
          type="checkbox"
        />
      </label>
      <label
        class="label-product"
      >
        Playstation
        <input
          class="checkbox"
          id="playstation"
          type="checkbox"
        />
      </label>
      <div
        class="btn-product-group"
      >
        <button
          class="btn-product"
          type="submit"
        >
          Submit
        </button>
        <button
          class="btn-product"
          type="submit"
        >
          Delete Card
        </button>
      </div>
    </form>
  </div>
</body>
`;
